pkg/util/ipvs/testing/fake.go

```diff
@@ -161,12 +178,12 @@ func (f *FakeIPVS) DeleteRealServer(serv *utilipvs.VirtualServer, dest *utilipvs
 	dests := f.Destinations[key]
 	var i int
 	for i = range dests {
-		if dests[i].Equal(dest) {
+		if toRealServerKey(dests[i]).String() == toRealServerKey(dest).String() {
 			break
 		}
 	}
 	// Not Found
-	if i >= len(f.Destinations[key]) {
+	if i >= len(f.Destinations[key])-1 {
```

Comment:
why change this line?

Reply:
Cause if `dests` don't have `dest`, i will be `len-1`. : (

Reply:
The modify seem incorrect to me - if last element of array match, what the value of `i` will be? I think it's `len-1`!

Reply:
Yeah,,, It's true, I have tested it, if no one match, it will also be `len-1`,,, maybe I should find other way,,,